src/ext/background.js

```diff
@@ -207,10 +216,44 @@ function processHeaders(details, url) {
  */
 function beforeSendHeaders(request, url) {
     // Cancel if we don't have a token to spend
-    if (!DO_REDEEM || !getSpendFlag(url.host) || checkMaxSpend(url.host) || spentUrl[url.href] || isErrorPage(url.href) || isFaviconUrl(url.href) || REDEEM_METHOD != "reload") {
-        return {cancel: false};
+
+    let reqUrl = url.href;
+    let host = url.host;
+
+    if (DO_REDEEM && !isErrorPage(reqUrl) && !isFaviconUrl(reqUrl)) {
+        // No reload method branch
+        if (REDEEM_METHOD === "no-reload") {
+            // check that we're at an URL that can handle redeems
+            const isRedeemUrl = SPEND_ACTION_URLS
+                .map(redeemUrl => patternToRegExp(redeemUrl))
+                .some(re => reqUrl.match(re));
+
+            if (countStoredTokens() > 0 && isRedeemUrl) {
+
+                const tokenToSpend = GetTokenForSpend();
+                if (tokenToSpend == null) {
+                    return {cancel: false};
+                }
+                setSpendFlag(host, null);
+                incrementSpentHost(host);
+
+                const http_path = request.method + " " + url.pathname;
+                const redemptionString = BuildRedeemHeader(tokenToSpend, url.hostname, http_path);
+                let headers = request.requestHeaders
+                headers.push({name: HEADER_NAME, value: redemptionString});
+                headers.push({name: HEADER_HOST_NAME, value: url.hostname});
+                headers.push({name: HEADER_PATH_NAME, value: http_path});
+                spendId[request.requestId] = true;
+                spentUrl[reqUrl] = true;
+                return {requestHeaders: headers};
+            }
+        } else if (REDEEM_METHOD === "reload" && getSpendFlag(host) && !checkMaxSpend(host) && !spentUrl[reqUrl]) {
+            return getReloadHeaders(request, url);
+        }
     }
-    return getReloadHeaders(request, url);
+
+    return {cancel: false}
+
```

Comment:
Tiny nit: could we remove the empty line here to be consistent with other functions?